github/Requester.py

```diff
@@ -51,19 +51,82 @@
 atLeastPython3 = sys.hexversion >= 0x03000000
 
 
+class RequestsResponse:
+    # mimic the httplib response object
+    def __init__(self, r):
+        self.status = r.status_code
+        self.headers = r.headers
+        self.text = r.text
+
+    def getheaders(self):
+        return self.headers.iteritems()
+
+    def read(self):
+        return self.text
+
+class HTTPSRequestsConnectionClass(object):
+    # mimic the httplib connection object
+    def __init__(self, host, port=None, strict=False, timeout=None, **kwargs):
+        self.port = port if port else 443
+        self.host = host
+        self.protocol = "https"
+        self.timeout = timeout
+        self.session = requests.Session()
+
+    def request(self, verb, url, input, headers):
+        self.verb = verb
+        self.url = url
+        self.input = input
+        self.headers = headers
+
+    def getresponse(self):
+        r = getattr(self.session,self.verb.lower())("%s://%s:%s%s" % (self.protocol, self.host, self.port, self.url), headers=self.headers, data=self.input, timeout=self.timeout)
+        return RequestsResponse(r)
```

Comment:
To make this more readable, I think it would better to use some temporary variables, like so:

```python
def getresponse(self):
    verb = getattr(self.session,self.verb.lower())
    url = "%s://%s:%s%s" % (self.protocol, self.host, self.port, self.url)
    r = verb(url, headers=self.headers, data=self.input, timeout=self.timeout)
    return RequestsResponse(r)
```

Same with `HTTPRequestsConnectionClass.getresponse` below.

Reply:
sure